File: app/src/main/java/ethiopia/covid/android/ui/widget/YekomeViewPager.java
package ethiopia.covid.android.ui.widget;


import android.annotation.SuppressLint;
import android.content.Context;
import android.util.AttributeSet;
import android.view.MotionEvent;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.viewpager.widget.ViewPager;

/**
 * Created by BrookMG on 4/11/2020 in ethiopia.covid.android.ui.widget
 * inside the project CoVidEt .
 */

public class YekomeViewPager extends ViewPager {
    private boolean isPagingEnabled = false;

    public boolean isPagingEnabled() {
        return isPagingEnabled;
    }

    public void setPagingEnabled(boolean pagingEnabled) {
        isPagingEnabled = pagingEnabled;
    }

    public YekomeViewPager(@NonNull Context context) {
        super(context);
    }

    public YekomeViewPager(@NonNull Context context, @Nullable AttributeSet attrs) {
        super(context, attrs);
    }

    @Override
    public boolean onInterceptTouchEvent(MotionEvent ev) {
        return isPagingEnabled && super.onInterceptTouchEvent(ev);
    }

    @SuppressLint("ClickableViewAccessibility")
    @Override
    public boolean onTouchEvent(MotionEvent ev) {
        return isPagingEnabled && super.onTouchEvent(ev);
    }
}

File: app/src/main/java/ethiopia/covid/android/ui/widget/YekomeViewPager.kt
package ethiopia.covid.android.ui.widget

import android.annotation.SuppressLint
import android.content.Context
import android.util.AttributeSet
import android.view.MotionEvent
import androidx.viewpager.widget.ViewPager

/**
 * Created by BrookMG on 4/11/2020 in ethiopia.covid.android.ui.widget
 * inside the project CoVidEt .
 */
class YekomeViewPager : ViewPager {
    var isPagingEnabled = false

    constructor(context: Context) : super(context)
    constructor(context: Context, attrs: AttributeSet?) : super(context, attrs)

    override fun onInterceptTouchEvent(ev: MotionEvent): Boolean {
        return isPagingEnabled && super.onInterceptTouchEvent(ev)
    }

    @SuppressLint("ClickableViewAccessibility")
    override fun onTouchEvent(ev: MotionEvent): Boolean {
        return isPagingEnabled && super.onTouchEvent(ev)
    }
}